[PCD8544] Replace project reference with reference to CharacterLcd NuGet package

MrCSharp22 opened the pull request:
This is to fix a missing reference runtime error when a project installs only the PCD8544 nuget.

## Description
- Added the `Iot.Device.CharacterLcd` nuget to the `PCD8544` project and the sample project.
- Added the `Iot.Device.CharacterLcd` nuget package reference to the `PCD8544` nuspec file.

## Motivation and Context
- Fixes nanoFramework/Home#1150

## How Has This Been Tested?
- Manually created a nuget package based on the new nuspec 
- Used the new package in a new project
- Deployed to the device and observed the error not occurring

## Types of changes
<!--- What types of changes does this PR introduce? Put an `x` in all the boxes that apply: -->
- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Config and build (change in the configuration and build system, has no impact on code or features)
- [x] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- PLEASE PLEASE PLEASE don't tick all of them just because -->
- [x] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [x] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

josesimoes commented:
@MrCSharp22 two things here: 
1. There are conflicts that you have to address when merging from upstream
2. The Pcd8544 project is referencing the CharacterLcd project directly, which means that all necessary code should be brought in when building the library and that wouldn't require referencing the CharacterLcd  nuget.

Can you please share the output with the message/error that you're getting?

MrCSharp22 commented:
@josesimoes 
Yup, I fixed the conflict already :) .
As for the error:

```
Resolving.
Link failure: some assembly references cannot be resolved!!

Assembly: NFApp1 (1.0.0.0) needs assembly 'Iot.Device.Pcd8544' (1.0.0.0)
Assembly: Iot.Device.Pcd8544 (1.0.0.0) needs assembly 'Iot.Device.CharacterLcd' (1.1.0.0)
Error: a3000000
Waiting for debug commands...
```

Now the nuspec file doesn't add the `Iot.Device.CharacterLcd` files (dll, pdbx, pe, and xml) directly in the `<files>` node. Do you think I should be doing that instead of using nuget?

MrCSharp22 commented:
@josesimoes 
Sorry, I deleted that comment by mistake. Restoring it here:

1- Yes, I fixed the conflict already.
2- This is the error from the debug output:

```
Resolving.
Link failure: some assembly references cannot be resolved!!

Assembly: NFApp1 (1.0.0.0) needs assembly 'Iot.Device.Pcd8544' (1.0.0.0)
Assembly: Iot.Device.Pcd8544 (1.0.0.0) needs assembly 'Iot.Device.CharacterLcd' (1.1.0.0)
Error: a3000000
Waiting for debug commands...
```

The nuspec file for `PCD8544` does not include `Iot.Device.CharacterLcd` (dll, pe, xml, and pdbx) in the list of files under the `<files>` node. Do you think I should do that instead of adding it via a nuget reference?

josesimoes commented:
@MrCSharp22 thanks! I now understood what's the error you're getting.
Following the same pattern as other IoT bindings that rely on other bindings, this fix of yours it's the way to go. 
Actually that's the best way to maintain the "cascade" update process.

josesimoes reviewed (approved):
LGTM! Thanks for taking care of this.

MrCSharp22 commented on devices/Pcd8544/samples/Pcd8544.samples/Pcd8544.samples.nfproj:
```diff
@@ -53,4 +53,4 @@
   </ProjectExtensions>
-</Project>
\ No newline at end of file
+</Project>
```
Sorry, I don't understand what is wrong here? I checked the file content and looks like all good?

josesimoes replied:
The sample project is still referencing the CharacterLcd project. 
If should reference the nuget as the Pcd8544 project is doing.
(I was under the impression that I've seen that on your initial commit, I could be wrong tough...)

MrCSharp22 replied:
I got it!
I didn't do that initially, But it is done now :) 

Thanks for flagging this one.
